src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,99 @@
+        String[] parts = input.split(" ");//split along the whitespace to get the integer
+        if (parts.length <= 1) {// checking for incomplete prompts
+            throw new DukeException();
+        }
+        String numStr = parts[1];
+        return Integer.valueOf(numStr);
+    }
+
+    public void execute() {
+        String input = sc.nextLine(); // scan the line for the user's input
+
+        boolean shouldContinue = true;
+        while (shouldContinue) {//create loop for the chat
+            try {
+                if (input.startsWith("find ")) {
+                    String desc = input.replaceFirst("find ", "");//remove find from input
+                    if (desc.isBlank()) {//check for incomplete input
+                        throw new DukeException();
+                    }
+                    list.find(desc);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("done ")) {
+                    int numInt = extractInt(input);
+                    list.setIndexDone(numInt);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("delete ")) {
+                    int numInt = extractInt(input);
+                    list.deleteTask(numInt);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("todo ")) {
+                    String task = input.replaceFirst("todo ", "");//remove to-do from input
+                    if (task.isBlank()) {//check for incomplete input
+                        throw new DukeException();
+                    }
+                    list.addToDo(task);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("deadline ")) {
+                    String task = input.replaceFirst("deadline ", "");//remove deadline from input
+                    if (task.isBlank()) {//checking incomplete input
+                        throw new DukeException();
+                    }
+                    String[] parts = task.split("/by ");//split string along keyword /by
+                    list.addDeadline(parts[0], parts[1]);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("event ")) {
+                    String task = input.replaceFirst("event ", "");//remove event from input
+                    if (task.isBlank()) {//checking incomplete input
+                        throw new DukeException();
+                    }
+                    String[] parts = task.split("/at");//split string along keyword /at
+                    list.addEvent(parts[0], parts[1]);
+                    input = sc.nextLine();
+                    continue;
+                }
+
+                switch (input) {
+                case "bye":
+                case "Bye":
+                case "BYE":
```

Comment:
According to the [coding standards](https://se-education.org/guides/conventions/java/intermediate.html#layout), the explicit `// Fallthrough` comment should be included whenever there is a case statement without a break statement. So perhaps you can add `// Fallthrough` at the end of line 79?